submissions/MarharytaBoichenko/oop/index.js

```diff
@@ -0,0 +1,102 @@
+class Inhabitant {
+  constructor(name, species, gender, friends, saying) {
+    this.name = name;
+    this.species = species;
+    this.gender = gender;
+    this.friends = friends;
+    this.saying = saying;
+    this.properties = ["name", "species", "gender", "friends", "saying"];
+  }
+  toPrint() {
+    return this.properties
+      .filter((prop) => {
+        return Boolean(this[prop]) != false;
+      })
+      .map((prop) => `my ${prop}: ` + this[prop])
+      .join(", ");
+  }
+}
+
+class Person extends Inhabitant {
+  constructor(name, gender, friends, saying) {
+    super(name, "human", gender, friends, saying);
+    this.hands = 2;
+    this.legs = 2;
+  }
+
+  toPrint() {
+    return super.toPrint() + `, my hands: ${this.hands}, my legs: ${this.legs}`;
+  }
+}
+
+class Man extends Person {
+  constructor(name, friends, saying) {
+    super(name, "man", friends, saying);
+  }
+
+  toPrint() {
+    return super.toPrint();
+  }
+}
+
+class Woman extends Person {
+  constructor(name, friends, saying) {
+    super(name, "woman", friends, saying);
+  }
+
+  toPrint() {
+    return super.toPrint();
+  }
+}
+class Animal extends Inhabitant {
+  constructor(name, species, paws, gender, friends, saying) {
+    super(name, species, gender, friends, saying);
+    this.paws = paws;
+  }
+
+  toPrint() {
+    return super.toPrint() + `, my paws: ${this.paws}`;
+  }
+}
+class Cat extends Animal {
+  constructor(name, species, paws, gender, friends) {
+    super(name, species, paws, gender, friends, "meow");
+  }
+
+  toPrint() {
+    return super.toPrint();
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender, friends) {
+    super(name, "dog", 4, gender, friends, "woof");
+  }
+
+  toPrint() {
+    return super.toPrint();
+  }
+}
+
+class Catwoman extends Cat {
+  constructor(name, paws, gender, hands, friends) {
+    super(name, "catwoman", paws, gender, friends);
+    this.hands = hands;
+  }
+
+  toPrint() {
+    return super.toPrint() + `, my hands: ${this.hands}`;
+  }
+}
```

Comment:
Notice, how hybrids break OOP principles: it forces you to duplicate properties.
Although, there are other approaches to achieve this without code duplication, the major learning here should be that when you need to compose properties and behaviours in a non-strictly-hiearchical manner, OOP won't work, and composition pattern is a way to go. Keep in mind and when you face a need to build hybrids, read about and use composition.